### DISCUSSION TITLE
Best way to send never matching refinement in query

### DISCUSSION THREAD

**Discussion by AntoineDuComptoirDesPharmacies** [algolia/instantsearch#5327] [Q&A] [answered]
Hi,

We are currently using React Instant Search to provide a superb search engine to our users.
In this engine, we propose a toggle checkbox that, when clicked, return only ads that are linked to its favorite products.

In order to implement this feature, we "store in fronted" user favorites as a list of product id. When user toggle the checkbox, we send a disjunctive refinement of this list on `product.id` attribute of ads. (using the same logic (reduce) as RefinementList connect)
This works well when user as at least one favorite.

However, as soon as he have no favorites yet, the reduce method will do nothing and so there will be no difference between checkbox toggled and untoggled. Our UX expectation is to have no ads returned when he is toggling the checkbox because he have no favorites yet.

In this situation, what is the best practice to reach this user experience ?
Maybe use a false refinement value of "-1" when there is no favorite yet to be sure all ads will be filtered out ?
An other idea ?

_Note :_ 
An other option would have been to add on each ads indexed in Algolia a field containing the list of users for which the ad is a favorite. Even if it will solve our current UX problem, we did not choose this option to avoid the large delay between client adding a new favorite and Algolia effective re-indexation.

Thanks in advance for your insights !
Yours faithfully,
LCDP

**Comment by dhayab** [comment, marked as answer]
Hi, I think it's a fine idea. You can indeed refine with an id you know won't exist in the records, this will return an empty list of results in this case.
